aodndata/soop/soop_co2_classifier.py

```diff
@@ -0,0 +1,129 @@
+from aodncore.pipeline import FileClassifier
+from aodncore.pipeline.exceptions import InvalidFileNameError, InvalidFileContentError, InvalidFileFormatError
+from ship_callsign import ship_callsign_list, ship_callsign
+import datetime
+
+VALID_PROJECT = ['IMOS', 'FutureReefMap', 'SOOP-CO2_RT']
+VESSEL_CODE = {'AA': 'VNAA',
+               'IN': 'VLMJ'}
+
+
+class SoopCo2FileClassifier(FileClassifier):
+
+    @classmethod
+    def def_project(cls, src_file):
+        """
+        Define project name according to file name or extension
+        Differentiate IMOS SOOP-CO2 RealTime from Delayed Mode and Future_Reef_MAP files
+        eg : (DM) IMOS_SOOP-CO2_GST_20121027T045200Z_VLHJ_FV01.nc
+             (DM) FutureReefMap_GST_20140530T185029Z_9V2768_FV01.nc
+             <Project_Name>_<<Facility-Code>>_<Data-Code>_<Start-date>_<Platform-Code>_FV<File-Version>.nc
+             (RT) IN_2017-165-0000dat.txt
+              <Vessel_code>_yyyy-ddd-hhmmdat.tx
+              :type src_file: object
+        """
+
+        if src_file.name.endswith('.nc'):
+            fields = cls._get_file_name_fields(src_file.name, min_fields=5)
+            if 'FV00' in src_file.name:
+                return fields[1] + '_RT'  # <Facility-Name + RTnc suffix>
+            else:
+                return fields[0]  # <Project-Name>
+        else:
+            raise InvalidFileFormatError(
+                "'{name}'is not a valid RT CO2 data (*dat.txt) or not a NetCDF file.".format(
```

Comment:
So far this is only handling '.nc' files. Is it also supposed to handle '*dat.txt'?

Reply:
it is handling the 'dat.txt' above